File: springboot-02-ioccontainer/src/main/java/com/linkedbear/springboot/assemble/a_module/selector/BarImportSelector.java
package com.linkedbear.springboot.assemble.a_b_c_module_profile_conditional.selector;

import com.linkedbear.springboot.assemble.a_b_c_module_profile_conditional.component.Bar;
import com.linkedbear.springboot.assemble.a_b_c_module_profile_conditional.config.BarConfiguration;
import org.springframework.context.annotation.ImportSelector;
import org.springframework.core.type.AnnotationMetadata;

public class BarImportSelector implements ImportSelector {
    
    @Override
    public String[] selectImports(AnnotationMetadata importingClassMetadata) {
        System.out.println("BarImportSelector invoke ......");
        return new String[] {Bar.class.getName(), BarConfiguration.class.getName()};
    }
}

File: springboot-02-ioccontainer/src/main/kotlin/com/linkedbear/springboot/assemble/a_module/selector/BarImportSelector.kt
package com.linkedbear.springboot.assemble.a_module.selector

import com.linkedbear.springboot.assemble.a_module.component.Bar
import com.linkedbear.springboot.assemble.a_module.config.BarConfiguration
import org.springframework.context.annotation.ImportSelector
import org.springframework.core.type.AnnotationMetadata

class BarImportSelector: ImportSelector {
    override fun selectImports(importingClassMetaData: AnnotationMetadata): Array<String> {
        println("BarImportSelector.selectImports() invoked ......")
        return arrayOf(Bar::class.java.name, BarConfiguration::class.java.name)
    }
}